modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/FilterToSqlHelper.java

```diff
@@ -126,9 +127,11 @@ class FilterToSqlHelper {
     Writer out;
     boolean looseBBOXEnabled;
     boolean encodeBBOXFilterAsEnvelope;
+    boolean pg12OrNewer;
 
-    public FilterToSqlHelper(FilterToSQL delegate) {
+    public FilterToSqlHelper(FilterToSQL delegate, boolean pg12OrNewer) {
```

Comment:
to keep backwards API compatibility you might want to keep the old constructor around with a default value of `false` for `pg12OrNewer`. eg.

```java
public FilterToSqlHelper(FilterToSQL delegate) {
        this.delegate = delegate;
        this.pg12OrNewer = false;
}
```
Perhaps with `@Deprecated` annotation

Reply:
I would also just get a Version field, otherwise these flags will profilerate... 10 years down the line we might have a pg120Newer, pg150Newer, pg180Newer and so on. A single Version that can be compared to a reference value is probably going to be better.

Reply:
Returned old constructor and changed parameter to Version